ci-operator/step-registry/ipi/aws/pre/edge-zones/opt-in/ipi-aws-pre-edge-zones-opt-in-commands.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+#
+# Select one random AWS Local Zone and Wavelength Zone each, opted into the
+# zone group (when opted-out), saving the zone name to be used in network
+# provisioning and install-config.yaml.
+#
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+export REGION="${LEASED_RESOURCE}"
+
+# Print a message with timestamp
+function echo_date() {
+    echo "$(date --rfc-3339=seconds)> $*"
+}
+
+function select_and_opt_in_zone() {
+    local zone_type=$1
+    local zone_name
+    local zone_group_name
+
+    # Randomly select the zone
+    zone_name=$(aws --region "$REGION" ec2 describe-availability-zones \
+        --all-availability-zones \
+        --filter Name=state,Values=available Name=zone-type,Values="${zone_type}" \
+        | jq -r '.AvailabilityZones[].ZoneName' | shuf | tail -n 1)
+    echo_date "Edge Zone selected: ${zone_name}"
+
+    zone_group_name=$(aws --region "$REGION" ec2 describe-availability-zones \
+        --all-availability-zones \
+        --filters Name=zone-name,Values="$zone_name" \
+        --query "AvailabilityZones[].GroupName" --output text)
+    echo_date "Zone Group discovered: ${zone_group_name}"
+
+    if [[ $(aws --region "$REGION" ec2 describe-availability-zones --all-availability-zones \
+            --filters Name=zone-type,Values="${zone_type}" Name=zone-name,Values="$zone_name" \
+            --query 'AvailabilityZones[].OptInStatus' --output text) == "opted-in" ]];
+    then
+        echo_date "Zone group ${zone_group_name} already opted-in"
+        echo "$zone_name" >> "${SHARED_DIR}/edge-zone-names.txt"
+        echo -en "$zone_group_name" > "${SHARED_DIR}/edge-zone-group-name.txt"
```

Comment:
Should save all zone groups?

```suggestion
        echo -en "$zone_group_name" >> "${SHARED_DIR}/edge-zone-group-name.txt"
```

Reply:
@yunjiang29 this file was created to keep compatibility with step `edge-nodes-machineset-manifests` used by QE workflows.
I believe we need to improve this saving like a map with the zone name, otherwise it will be complicated to further steps. My suggestion is to change this file to something like:
1) create one zone group file by zone name. Example: `${SHARED_DIR}/edge-zone-group-name_${zone_name}.txt`; OR
2) create a single file with array indexing by zone name. Example:
```sh
# init
echo "declare -A zone_groups" > "${SHARED_DIR}/edge-zone-groups.env"

# for each discovered zone
echo "zone_groups[$zone_name]=\"$zone_group_name\"" >> "${SHARED_DIR}/edge-zone-groups.env"

# consumers (eg step edge-nodes-machineset-manifests)
source "${SHARED_DIR}/edge-zone-groups.env"
# use "${zone_groups[$zone_name]}"
```

WDYT?

Reply:
> to keep compatibility with step `edge-nodes-machineset-manifests` used by QE workflows.

Thank you @mtulio, I prefer option 2.

Reply:
@yunjiang29 moved to the option 2, and the affected step has been updated.

Reply:
Thanks!